### DISCUSSION TITLE
Debian Testing as a rolling release?

### DISCUSSION THREAD

**Discussion by nomad101** [SpiralLinux/SpiralLinux-project#462] [General]
Hi, 

I've used linux for a few years, though still a newbie. I'm thinking to use Spiralinux and set it to testing reps. This could effectively act as rolling release, so I wouldn't need to reinstall software. And because there are snapshots if breakages occur while updating, I could roll-back. Would I have a fairly stable system for a home user that could be used as a daily driver?

I'm actually not that concerned about using the most bleeding edge software. More about not having to do a clean install of spiralinux after a major version upgrade of Debian. I saw that there are [upgrade instructions for Spiralinux across versions](https://github.com/SpiralLinux/SpiralLinux-project/wiki) on doing this, but is it liable to cause problems down the line? Elsewhere I've read that it's normally recommended to do a clean install of debian. 

Essentially, if I want to have a rolling release for the main reason of not having to reinstall all my packages again, should I use testing or stick with stable, and follow the upgrade instructions?

Thanks for any thoughts and advice.

**Comment by AlphaElwedritsch** [comment]
In the end, testing is not stable. Even if it is less common for Debian to destroy your system, it can happen. It has happened to me too.

A change from stable to the next one usually works more smoothly with Debian than with other distros. But that can also go wrong.

So there is no standard recipe. Unfortunately.

I, but I am only speaking for myself here, with no guarantees, would stay on the stable and do a dist-upgrade every two years. In my opinion, that is the least risky option

**Comment by geckolinux** [comment]
Hi there, you'll never have to do a clean install of SpiralLinux to upgrade it, and the same applies for a vanilla Debian Stable installation as well. Upgrades from one Debian Stable release to the next one are extremely reliable. Debian perfected in-place release upgrades long before anybody else, and arbitrary recommendations of performing a clean install without a specific reason make no sense.

As for the Debian Testing branch, as its name implies it's not a "release", but rather a branch that is used during the development process with the ultimate goal of reaching a Stable release. As such, it may or may not work better than whatever distro you're coming from.

And just a reminder that SpiralLinux by design does not have its own repositories or upgrade mechanisms, so its upgrade process and characteristics when running the Stable release vs Testing are exactly the same as a vanilla Debian installation. However, SpiralLinux does have very different default settings compared to vanilla Debian, giving users a safety net of snapshot rollbacks in case an upgrade or Testing / Unstable update causes problems, which is much more time consuming to configure manually in a vanilla Debian installation.

**Reply by nomad101** [reply]
I've only ever used other debian-based distros and the advice has always been to do a clean re-install when there's a new debian release. 
After all this time, I hadn't realised that vanilla debian stable can be upgraded to its next major version without having to reinstall packages again! And now I better understand the upgrade instructions for Spiralinux. 

Thank-you for your reply, and for creating Spiralinux :+1: 

EDIT: while this is based on selected users, interesting to read: [reddit thread](https://www.reddit.com/r/debian/comments/13pmcgl/with_debian_12_just_around_the_corner_would_you/)

**Reply by geckolinux** [reply]
Hi there,

> other debian-based distros

I think the key word here is Debian _based_ distros. Many distros take Debian packages and mix them with their own custom packages and repositories that are not part of the system of packages that Debian officially tests to work when upgrading from one Stable release to the next. That's one of the reasons why SpiralLinux directly uses unmodified Debian packages from the official repositories. It's not a Debian _based_ distro, in fact it's not even a distro, it's a spin that offers a different package selection and installation method and a different configuration from vanilla Debian, but it's all within the realm of what an experienced user could install and (tediously) configure manually with the official Debian installation media.

As for upgrading a vanilla Debian Stable installation, they include detailed instructions in the release notes for each CPU arch, for example: https://www.debian.org/releases/stable/amd64/release-notes/ch-upgrading.en.html
